### PR TITLE
minitest: add lite-sata minitest to verify GTP status

### PR DESCRIPTION
**Pull request by acomodi**
Signed-off-by: Alessandro Comodi <acomodi@antmicro.com>

This PR is to add a LiteX minitest that makes use of liteSATA, which implements the GTP-related tiles

### CODE REVIEW

**Comment by mithro** [comment]
@acomodi - MIght be worth looking at https://github.com/hamsternz/DisplayPort_Verilog

**Comment by acomodi** [comment]
@mithro Sure, that is yet another good candidate for a minitest

**Review by litghost — CHANGES_REQUESTED** [review]
This minitest needs a README?

File: minitests/litex/nexys_video_sata/Makefile [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+clean:
+	@find . -name "build-par.*" | xargs rm -rf
+	@find . -name "build-syn.*" | xargs rm -rf
+	@rm -f *.edif
+	@rm -f *.bit
+	@rm -f *.bin
+	@rm -f *.log
+	@rm -f *.dcp
+
+help:
```
Comment (litghost) [inline minitests/litex/nexys_video_sata/Makefile:25]: Looks like the "all" target is missing:
```
$ make help
Usage: make all [SYNTH=<vivado/yosys>]
$ make all SYNTH=vivado
make: *** No rule to make target 'all'.  Stop.
```

File: minitests/litex/nexys_video_sata/Makefile [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+# SPDX-License-Identifier: ISC
+
+SHELL = bash
+
+SYNTH ?= vivado
+YOSYS = $(XRAY_DIR)/third_party/yosys/yosys
+PART = xc7a200tsbg484-1
+PROJECT_NAME = sata
+
+clean:
```
Comment (litghost) [inline minitests/litex/nexys_video_sata/Makefile:16]: The first target in a Makefile is the default target.  I don't think `clean` is a good default target.  Recommend putting the `all` target first?  If target ordering matters, you can use `.DEFAULT_GOAL` (see https://www.gnu.org/software/make/manual/html_node/Special-Variables.html#Special-Variables)

**Comment by acomodi** [comment]
> This minitest needs a README?

Indeed, I forgot to put this as WIP.

I'll need to also fix the litex modules to have a reproducible test and also fix some issues in the Makefile.

**Review by litghost — CHANGES_REQUESTED** [review]
Looks like the path to the venv is assumed in the makefile.  For example, if this makefile is invoked used the prjxray venv, you get the following error:
```
./src/litex-boards/litex_boards/targets/nexys_video.py --with-sata --integrated-rom-size 0x10000
bash: line 1: ./src/litex-boards/litex_boards/targets/nexys_video.py: No such file or directory
make: *** [Makefile:38: build/build.ok] Error 127
```

File: minitests/litex/nexys_video_sata/README.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+The minitest synthesis step can be performed with Yosys or Vivado.
+
+The final FASM file with the `unknown bits` can be obtained by running the following:
+
+```bash
+make all
+```
+
+All the pre-requisites (LiteX, Yosys, etc.) are automatically installed/built. It is required though to have Vivado installed in the system.
```
Comment (litghost) [inline minitests/litex/nexys_video_sata/README.md:29]: This project requires a RISC-V tool chain on the path, so this is not true?

Comment (acomodi) [reply minitests/litex/nexys_video_sata/README.md:29]: Fixed in Makefile

**Pull request merged**